src/handlers/base_handler.py

```diff
@@ -0,0 +1,85 @@
+from queue import Queue
+
+from bson import ObjectId
+from tornado_json.requesthandlers import APIHandler
+
+import src.constants.api_keys as api_key
+import src.constants.constants as consts
+import src.constants.db_keys as db_key
+from src.auth import validate_id
+from src.config import BAD_REQUEST_CODE
+from src.database import DatabaseResolver
+from src.utilities import get_time, resolve_env_vars
+
+
+class BaseAPIHandler(APIHandler):
+    def abort(self, data, status):
+        self.set_status(status)
+        self.fail(data)
+
+    def get_db(self):
+        # type: () -> DatabaseResolver
+        return self.settings.get(consts.APP_DB)
+
+    def get_queue(self):
+        # type: () -> Queue
+        return self.settings.get(consts.APP_QUEUE)
+
+    def get_cluster_token(self):
+        # type: () -> str
+        return self.settings.get(consts.APP_TOKEN)
+
+    @validate_id
```

Comment:
Is there any particular reason that the three decorators need to be decorators? It add additional complexity because the validation parts are heavily coupled with the function param definition i.e. if id_ isn't the second parameter then this entire thing breaks.

It could also be a function call which would keep the semantics of `self` cleaner.

Reply:
This seems to be the canonical way doing middleware-esque things. 

Are you suggesting calling `self.whatever` each time you want to do some sort of middleware task? That seems unclean and not pythonic to me.

Reply:
I'm completely fine with middleware decorators. The problem is that the deocrators themselves are tightly coupled with the implementation of the function and the parameter names.

My general philosophy is that any decorator ought to be able to be substituted to any other decorator. My cannonical director for http-related function is `@404onException` which treats the function as black box and if an executor raises an exception then do something.

With `validate_id` The definition has the following two lines

```
self = ...
id_ = kwargs.get("id_") if len(args) < 2 else args[1]
```

Which means that `id_` if its name gets changed immediately breaks a portion of the code.

You can fix this problem with decorators too, but it leads to another layer of indirection

```
@validate_id('id_')
def foo(self, id_):
```

I guess this is my usually django experience is to ditch the entire thing and keep it clean and simple (pythonic is a hotly debated word, so I'll fall back on [Guido's mantra of for-loops](https://www.quora.com/Why-did-Guido-van-Rossum-want-to-remove-lambda-from-Python-3) and err on the side of readable).

```
def foo(self, id_):
    if not validate_id(id_):
        return ...
```

Reply:
I understand your point too. I used them because they seemed more intuitive and cleaner to me. I am rather conflicted what is the _correct_ way to go here.

Yes, these decorators are very coupled with the functions they are used across. I did not feel the need to make them generic enough while developing them.

Reply:
I removed `validate_id` and `autheticate_worker` decorators because of the reasons mentioned above. Passing a parameter into the decorator like:
```
@validate_id('id_')
def foo(self, id_):
```
is really not worth the increase in complexity. We will have to have another level of nested functions like:
```
def decorator(argument):
    def real_decorator(function):
        def wrapper(*args, **kwargs):
            stuff()
        return wrapper
    return real_decorator
```

I am keeping the `autheticate` decorator because it is not tied to any parameter name as such and is kinda clean.